src/openrct2/drawing/Drawing.Sprite.cpp

```diff
@@ -595,10 +595,10 @@ void FASTCALL gfx_sprite_to_buffer(DrawSpriteArgs& args)
  *
  *  rct2: 0x00681DE2
  */
-void FASTCALL
-    gfx_draw_sprite_raw_masked_software(rct_drawpixelinfo* dpi, int32_t x, int32_t y, int32_t maskImage, int32_t colourImage)
+void FASTCALL gfx_draw_sprite_raw_masked_software(
+    rct_drawpixelinfo* dpi, const ScreenCoordsXY& scrCoords, int32_t maskImage, int32_t colourImage)
 {
-    int32_t left, top, right, bottom, width, height;
+    int32_t left, top, right, bottom, width, height, x = scrCoords.x, y = scrCoords.y;
```

Comment:
Don't declare them here, declare an actual `ScreenCoordsXY` object that takes them both on lines 626 and 627. Something like:
```
auto offsetCoords = scrCoords + ScreenCoordsXY{ imgMask->x_offset, imgMask->y_offset };
```

Please also use `scrCoords` directly on line 612, that calls `gfx_draw_sprite_software`, instead of constructing another object from `{ x, y }` that is exactly what `scrCoords` is.